cmd/multi-cluster/manifests/leaf-cluster/developers-rbac.yaml

```diff
@@ -0,0 +1,31 @@
+# Defining developers group access
+# How we can impersonate it:
+# grab the service account token and set it to a user
+# kubectl config set-credentials sa-user --token=$(kubectl get secret default-token-p6kj6 -o jsonpath={.data.token} | base64 -d)
```

Comment:
@luizbafilho  I'm guessing the secret you are referring to here is the one in the `developers` namespace?

Reply:
I see different secrets with similar name in many different namespaces 🤔 
<img width="380" alt="image" src="https://user-images.githubusercontent.com/3892786/157315599-25bd5066-e013-4d83-873c-c3e9bf7168e5.png">

Reply:
that would be the one in flux-system which is I'm doing the binding https://github.com/weaveworks/weave-gitops/pull/1556/files#diff-63fa25e919d6252750928dd7f73dfc4da5e7388073361c9ed8dbcb9d557919fbR16

Reply:
> that would be the one in flux-system which is I'm doing the binding
Then the command should be:
```
kubectl config set-credentials sa-user --token=$(kubectl get secret -n flux-system $(kubectl get secret -n flux-system | grep -o "default-token-[^ ]*") -o jsonpath='{.data.token}' | base64 -d)
```

Reply:
As a workaround I followed the next steps:
1) Deployed a nginx pod mounting the service account for impersonation
2) Started a bash session within that pod
3) Setup CACERT and TOKEN using the service account credentials
3) Run this command :
```
curl --cacert $CACERT --header "Authorization: Bearer $TOKEN" -header "Impersonate-User: luiz.filho@weave.works" --header "Imporsonate-Group: developers" -X GET https://kubernetes.default.svc/api/v1/namespaces/developers/pods
{
  "kind": "PodList",
  "apiVersion": "v1",
  "metadata": {
    "resourceVersion": "30365"
  },
  "items": []
}root@nginx-5d989b88db-dxzkk:/#
```
And it works perfect.

Reply:
great, setting that up is tricky any bit wrong and nothing works, to help me out I've created a small tool, that wont be needed anymore since on core we'll be using just the underlying connection but here it is: https://github.com/weaveworks/weave-gitops/blob/0c0a7efecf2a4dc7e0db706ba6a08c6dad70ad92/tools/add-cluster/main.go

Reply:
I was mostly trying to understand and put into practice the concept of impersonation